remote/remote_test.go

```diff
@@ -168,3 +171,84 @@ func doTestWatch(t *testing.T, sm subnet.Manager) {
 		t.Errorf("WatchSubnet produced wrong subnet: expected %s, got %s", l.Key(), evt.Lease.Key())
 	}
 }
+
+func doTestWatchNetworks(t *testing.T, sm subnet.Manager, serverRegistry *subnet.MockSubnetRegistry) {
+	ctx, cancel := context.WithCancel(context.Background())
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	defer func() {
+		cancel()
+		wg.Wait()
+	}()
+
+	// Because remote clients are not allowed to add/remove networks, there
+	// is no server API for these operations.  But testing network watches
+	// still requires this, and that must happen directly in the server's
+	// registry.  That can cause race conditions if the WatchNetworks
+	// goroutine hasn't been run yet by the time CreateNetwork is called,
+	// so testing operations that touch the server registry must be deferred
+	// a short time to ensure that the server is listening when they happen.
+
+	events := make(chan []subnet.Event)
+	go func() {
+		subnet.WatchNetworks(ctx, sm, events)
+		wg.Done()
+	}()
+
+	// skip over the initial snapshot
+	<-events
+
+	expectedNetname := "foobar"
+	go func() {
+		// Add a new network a short time later to ensure that the
+		// client-side network watch triggers a server watch before the
+		// network is added directly in the server with CreateNetwork()
+		time.Sleep(time.Second)
```

Comment:
I don't think I fully understand the need for sleep. Is the desire to exercise the code where the server does an etcd watch prior to the network key being created? In that case I guess it's ok but the code should work regardless. That is, this also needs to work:

1. Client side issues a watch
2. A new network is created in etcd
3. The server receives the watch and processes it by immediately returning
4. The client gets notified of the new network